projects/hathor-wallet-mobile/0002-nano-contract-integration.md

```diff
@@ -0,0 +1,476 @@
+
+The header section of this screen provides essential state information about the selected *Nano Contract*, including the full *Nano Contract ID*, *Blueprint Name*, and a breakdown of each *Token Balance*. This header is designed to be both expandable and collapsible: in its expanded form, it reveals all state details, while in its collapsed form, it displays only the *Nano Contract ID* for a more streamlined view.
+
+The list is encapsulated by the *Nano Contract Transaction List* component.
+
+##### Design Bill
+- **Placement**:
+	- Position the header with state information at the top of the screen.
+	- Place the *Nano Contract Transaction List* component below the header.
+
+- **Functionality**:
+	- The header should display the full *Nano Contract ID*, *Blueprint Name*, and *Token Balances*.
+
+- **Visibility Conditions**:
+	- Include a toggle in the header for expanding and collapsing to show/hide detailed state information.
+
+### Component: Nano Contract Transaction List
+[component-nano-contract-transaction-list]: #component-nano-contract-transaction-list
+
+This component displays a list of *Nano Contract Transaction* items. Each item represents a transaction as either an input (sent), an output (received), or a method execution (called). The list is designed to provide essential transaction information in a clear and concise manner.
+
+Tapping on an item opens the *Nano Contract Transaction Details* component, providing further information on the selected transaction item.
+
+##### Design Bill
+- **Placement**:
+	- The *Nano Contract Transaction List* should be displayed within the body section of the *Nano Contract Transactions* screen.
+
+- **Functionality**:
+	- Each item in the list is actionable, leading to detailed transaction information upon selection.
+	- Implement an infinite scrolling view to display a potentially large number of transactions.
+	- Each transaction item must contain *Description*, *Token Amount*, *Token Symbol*, *Blueprint Method Name* and *Date and Time* information for the item list view.
+
+- **Visibility Conditions**:
+	- Differentiate item types *sent*, *received*, and *called* using distinct visual indicators.
+	- A *called* item type must not contain *Token Amount* and *Token Symbol* information. 
+
+>[!NOTE]
+>Example of item description per type:
+>- "You *received* `<Token Symbol>` from `<Blueprint Method Name>`"
+>- "You *sent* `<Token Symbol>` to `<Blueprint Method Name>`"
+>- "You *called* `<Blueprint Method Name>`"
+
+### Component: Nano Contract Transaction Details
+
+This component provides detailed information for each transaction item type: *sent*, *received*, and *called*.
+
+For *sent* and *received* transaction items, the header displays *Token Amount* and *Token Symbol*, with an optional label *Amount* underneath them.
+
+The body section includes:
```

Comment:
Should we also include information about the method call? Like which parameters were passed and the method call result?

Reply:
> Should we also include information about the method call?

Yes, we can include the arguments called.

> Like which parameters were passed and the method call result?

Public methods are always void.

Reply:
I've discussed this with Pedro and we came to the conclusion to redirect users to explorer website on the trasaction page, where a user will be able to see all the details.